src/main/java/org/testeditor/web/xtext/index/XtextIndexApplication.xtend

```diff
@@ -55,38 +54,27 @@ class XtextIndexApplication extends Application<XtextIndexConfiguration> {
 		bootstrap.objectMapper.registerModule(customSerializerModule)
 	}
 
-	/**
-	 * please override and provide own (language dependent) injected xtext index instance
-	 */
-	protected def XtextIndex getIndexInstance() {
-		val injector = Guice.createInjector(#[new XtextIndexModule])
-		return injector.getInstance(XtextIndex)
+	override run(XtextIndexConfiguration configuration, Environment environment) {
+		super.run(configuration, environment)
+		configureServices(configuration, environment)
 	}
 
+	abstract protected def List<ISetup> getLanguageSetups()
+
+	abstract protected def XtextIndex getIndexInstance()
```

Comment:
This puts the burden of instantiating the index on implementors; it should be the responsibility of this class / project.